src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name, health) {
+    super(name, health);
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore && target.hidden === false) {
+      target.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter((a) => a.health > 0);
```

Comment:
[CODE PERFORMANCE] - While filtering the `Animal.alive` array inside the `bite` method is acceptable, it's better to do this after checking if the target's health is less than or equal to 0. This avoids unnecessary filtering operations if no animal's health has reached 0.